app/main/api/handlers/getWallet.ts

```diff
@@ -0,0 +1,104 @@
+import { LiteralType } from "io-ts"
+
+import { asRuntimeType, asObject } from "app/common/runtimeTypes"
+import { SubSystems } from "app/main/system"
+import { JsonSerializable } from "app/common/serializers/json"
+import { WalletInfo, Wallet } from "app/common/runtimeTypes/storage/wallets"
+import { Storage } from "app/main/storage"
+import { inject } from "app/main/utils/utils"
+import {
+  GetWalletError, GetWalletParams, GetWalletResponse, GetWalletSucccess, getWalletErrors,
+  GetWalletErrors
+} from "app/common/runtimeTypes/ipc/wallets"
+
+/**
+ * Handler function for "getWallet" method.
+ *
+ * @export
+ * @param {SubSystems} system
+ * @param {GetWalletParams} params
+ * @returns {Promise<GetWalletResponse>}
+ */
+const getWallet = async (system: SubSystems, params: any): Promise<JsonSerializable> =>
+  Promise.resolve({ params })
+    .then(parseParams)
+    .then(inject(checkWalletInfo, system))
+    .then(inject(replaceStorage, system))
+    .then(searchWallet)
+    .then(encodeResponse)
+    .catch(buildError)
```

Comment:
In no way will this function return anything `JsonSerializable` unless the result is effectively encoded with `asObject()`.

![The cake is a lie!](https://img00.deviantart.net/a90a/i/2011/311/6/3/portal___the_cake_is_a_lie__by_caycowa-d4f8x5o.png)

Reply:
:cake: